crates/sui-indexer/tests/integration_tests.rs

```diff
@@ -387,6 +430,156 @@ pub mod pg_integration_test {
+                Some(show_some_content.clone().with_content()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_owner()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_type()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_display()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_bcs()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_previous_transaction()),
+            ),
+        ];
+
+        let results: Vec<SuiObjectResponse> = join_all(futures)
+            .await
+            .into_iter()
+            .collect::<Result<_, _>>()
+            .unwrap();
+
+        let expected_results = vec![
+            // bcs_lossless
+            SuiObjectData {
+                display: None,
+                content: None,
+                ..post_transfer_full_obj_data.clone()
+            },
+            // full_content
+            SuiObjectData {
+                bcs: None,
+                display: None,
+                ..post_transfer_full_obj_data.clone()
+            },
+            // non-optional
+            object_required_fields.clone(),
+            SuiObjectData {
+                content: post_transfer_full_obj_data.content.clone(),
+                ..object_required_fields.clone()
+            },
+            SuiObjectData {
+                owner: post_transfer_full_obj_data.owner,
+                ..object_required_fields.clone()
+            },
+            SuiObjectData {
+                type_: post_transfer_full_obj_data.type_.clone(),
+                ..object_required_fields.clone()
+            },
+            SuiObjectData {
+                display: post_transfer_full_obj_data.display.clone(),
+                ..object_required_fields.clone()
+            },
+            SuiObjectData {
+                bcs: post_transfer_full_obj_data.bcs.clone(),
+                ..object_required_fields.clone()
+            },
+            SuiObjectData {
+                previous_transaction: post_transfer_full_obj_data.previous_transaction,
+                ..object_required_fields.clone()
+            },
+        ];
+
+        for (received, expected) in results.iter().zip(expected_results.iter()) {
+            let data = received.object()?;
+            assert_eq!(data, expected);
+            assert_ne!(data.version, initial_full_obj_data.version);
+            assert_eq!(data.object_id, initial_full_obj_data.object_id);
+        }
+
+        // deleted(SuiObjectRef);
```

Comment:
pending figuring out how to delete an object https://mysten-labs.slack.com/archives/C04HS54LHUM/p1679353365380589

Reply:
per Patrick's comment, we can merge coin to delete one.

Reply:
hehe thanks all